datacube/storage/_base.py

```diff
@@ -42,11 +42,10 @@ def _get_band_and_layer(b: Dict[str, Any]) -> Tuple[Optional[int], Optional[str]
 
 
 def _extract_driver_data(ds: Dataset) -> Optional[Any]:
-    dd = getattr(ds, 'driver_data', None)
-    if dd is None:
-        dd = getattr(ds, 's3_metadata', None)  # TODO: change CSIRO driver to populate `driver_data`
-    return dd
-
+    if 'storage_metadata' in ds.metadata_doc:
```

Comment:
1. Let's keep `driver_data` key, I think `driver data` communicates more clearly that this data is not interpreted by datacube itself and is only passed on to the driver for interpretation.

2. Can you also please change it to `return ds.metadata_doc.get('driver_data', None)` to reduce need for testing branches, it is also a safer paradigm for multi-threaded applications.

Reply:
Done